feature/savings/src/commonMain/kotlin/org/mifos/mobile/feature/savings/savingsAccountApplication/SavingsAccountApplicationViewModel.kt

```diff
@@ -12,55 +12,59 @@ package org.mifos.mobile.feature.savings.savingsAccountApplication
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.ExperimentalCoroutinesApi
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
 import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.stateIn
 import kotlinx.coroutines.launch
 import org.mifos.mobile.core.common.Constants
-import org.mifos.mobile.core.common.utils.DateHelper
-import org.mifos.mobile.core.common.utils.getTodayFormatted
+import org.mifos.mobile.core.common.DataState
+import org.mifos.mobile.core.common.DateHelper
 import org.mifos.mobile.core.data.repository.SavingsAccountRepository
-import org.mifos.mobile.core.datastore.PreferencesHelper
+import org.mifos.mobile.core.datastore.UserPreferencesDataSource
 import org.mifos.mobile.core.model.entity.accounts.savings.SavingsAccountApplicationPayload
 import org.mifos.mobile.core.model.entity.accounts.savings.SavingsAccountUpdatePayload
 import org.mifos.mobile.core.model.entity.accounts.savings.SavingsWithAssociations
 import org.mifos.mobile.core.model.entity.templates.savings.SavingsAccountTemplate
 import org.mifos.mobile.core.model.enums.SavingsAccountState
-import org.mifos.mobile.feature.savings.R
 import org.mifos.mobile.feature.savings.savingsAccountApplication.SavingsAccountApplicationUiState.Loading
-import javax.inject.Inject
 
-@HiltViewModel
-internal class SavingsAccountApplicationViewModel @Inject constructor(
+internal class SavingsAccountApplicationViewModel(
     private val savingsAccountRepositoryImp: SavingsAccountRepository,
-    private val preferencesHelper: PreferencesHelper,
+    private val preferencesHelper: UserPreferencesDataSource,
     savedStateHandle: SavedStateHandle,
 ) : ViewModel() {
 
     private val clientId get() = preferencesHelper.clientId
 
     private val savingsId =
         savedStateHandle.getStateFlow(key = Constants.SAVINGS_ID, initialValue = -1L)
+
     private val savingsAccountState = savedStateHandle.getStateFlow(
         key = Constants.SAVINGS_ACCOUNT_STATE,
-        initialValue = SavingsAccountState.CREATE,
-    )
+        initialValue = SavingsAccountState.CREATE.name,
+    ).value.let { SavingsAccountState.valueOf(it) }
```

Comment:
@revanthkumarJ that's why I said before you changed `enum` to `string` and now back to enum. So fix it when passing it from navigation.